elephant/phase_analysis.py

```diff
@@ -82,107 +92,77 @@ def spike_triggered_phase(hilbert_transform, spiketrains, interpolate):
-            # Difference vector between actual spike time and sample point,
-            # positive if spike time is later than sample point
-            dv = spiketrain[sttimeind[spike_i]] - times[ind_at_spike_j]
-
-            # Make sure ind_at_spike is to the left of the spike time
-            if dv < 0 and ind_at_spike_j > 0:
-                ind_at_spike_j = ind_at_spike_j - 1
+    for spiketrain_i in range(num_spiketrains):
+        full_sp = spiketrains[spiketrain_i]
+        for trial_i in range(num_trials):
+            spike_times = full_sp.time_slice(t_starts[trial_i], t_stops[trial_i]).times  # leave out last spike
+            time = times[trial_i]
+            phase = phases[trial_i]
+            amp = amps[trial_i]
+
+            # cut righthand border
+            if spike_times[-1] == time[-1]:
+                spike_times = spike_times[:-1]
 
             if interpolate:
-                # Get relative spike occurrence between the two closest signal
-                # sample points
-                # if z->0 spike is more to the left sample
-                # if z->1 more to the right sample
-                z = (spiketrain[sttimeind[spike_i]] - times[ind_at_spike_j]) /\
-                    hilbert_transform[phase_i].sampling_period
-
-                # Save hilbert_transform (interpolate on circle)
-                p1 = np.angle(hilbert_transform[phase_i][ind_at_spike_j])
-                p2 = np.angle(hilbert_transform[phase_i][ind_at_spike_j + 1])
-                result_phases[spiketrain_i].append(
-                    np.angle(
-                        (1 - z) * np.exp(np.complex(0, p1)) +
-                        z * np.exp(np.complex(0, p2))))
-
-                # Save amplitude
-                result_amps[spiketrain_i].append(
-                    (1 - z) * np.abs(
-                        hilbert_transform[phase_i][ind_at_spike_j]) +
-                    z * np.abs(hilbert_transform[phase_i][ind_at_spike_j + 1]))
+                # use the scipy.interp1d fits to get phases at spiketimes
+                phase_fit = interp1d(time, phase, kind='linear')
+                amp_fit = interp1d(time, amp, kind='linear')
+                result_phases[spiketrain_i].append(phase_fit(spike_times))
+                result_amps[spiketrain_i].append(amp_fit(spike_times))
+                result_times[spiketrain_i].append(spike_times)
             else:
-                p1 = np.angle(hilbert_transform[phase_i][ind_at_spike_j])
-                result_phases[spiketrain_i].append(p1)
-
-                # Save amplitude
-                result_amps[spiketrain_i].append(
-                    np.abs(hilbert_transform[phase_i][ind_at_spike_j]))
 
-            # Save time
-            result_times[spiketrain_i].append(spiketrain[sttimeind[spike_i]])
+                # for each spike, get index in time closest to spiketime
+                spike_phase_idx = [np.argmin(np.abs(time - sptime)) for sptime in spike_times]
```

Comment:
I'll update this. I think it would be better to rescale + convert to pure array before doing any further computations.

Reply:
@dizcza Thanks for starting the review. Regarding the pep8/style points, I'd say lets do this last, after we find a proper format and checked all the code (as you said).

I'll add myself as author once its finished :). 

I don't see any comment from @mdenker yet (am I missing something?). Regarding input I don't have any specific requirements myself, this felt intuitive for me and figured I could try and think really hard what's the best format for elephant, but that's actually something you guys are much better at. Just let me know any thoughts and I'm sure I'll be fixed.

Then for the code itself, the one thing I haven't check is how the interpolation performs on circular data.

Finally the main reason why I'm starting this was because I'd like to see 'Pairwise Phase Consistency'  and maybe a few similar methods. They need the same base computations as STP (ie angle per spike) which is why I think it can be a natural extention of this module. Either by extending this method are adding other methods. [Fieldtrip is my inspiration here](http://www.fieldtriptoolbox.org/reference/ft_spiketriggeredspectrum_stat/), it also has the proper references to the publications. If you think this should be here, let me know (ideally also in what form).